tests/compiler/std/string.ts

```diff
@@ -731,10 +733,118 @@ assert(dtoa(0.000035689) == "0.000035689");
+assert(
+  "𠜎 𠜱 𠝹 𠱓 𠱸 𠲖 𠳏 𠳕 𠴕 𠵼 𠵿 𠸎 𠸏 𠹷 𠺝 𠺢 𠻗 𠻹 𠻺 𠼭 𠼮 𠽌 𠾴 𠾼 𠿪 𡁜 𡁯 𡁵 𡁶 𡁻 𡃁"
+  .toUpperCase().toLowerCase() ==
+  "𠜎 𠜱 𠝹 𠱓 𠱸 𠲖 𠳏 𠳕 𠴕 𠵼 𠵿 𠸎 𠸏 𠹷 𠺝 𠺢 𠻗 𠻹 𠻺 𠼭 𠼮 𠽌 𠾴 𠾼 𠿪 𡁜 𡁯 𡁵 𡁶 𡁻 𡃁"
+);
+
+assert(String.fromCodePoint(0x10000).toLowerCase() == "𐀀");
+assert(String.fromCodePoint(0x10000).toUpperCase() == "𐀀");
+
+// Tests some special casing for lower case mapping
+assert("\u1F88".toLowerCase() == "\u1F80");
+assert("\u1F8F".toLowerCase() == "\u1F87");
+assert("\u1FFC".toLowerCase() == "\u1FF3");
+
+// Tests some special casing for upper case mapping
+assert("\uFB00".toUpperCase() == "FF");
+assert("\uFB01".toUpperCase() == "FI");
+assert("\uFB02".toUpperCase() == "FL");
+assert("\uFB03".toUpperCase() == "FFI");
+assert("\uFB04".toUpperCase() == "FFL");
+assert("\uFB05".toUpperCase() == "ST");
+assert("\uFB06".toUpperCase() == "ST");
+assert("\u01F0".toUpperCase() == "J\u030C");
+assert("\u1E96".toUpperCase() == "H\u0331");
+assert("\u1E97".toUpperCase() == "T\u0308");
+assert("\u1E98".toUpperCase() == "W\u030A");
+assert("\u1E99".toUpperCase() == "Y\u030A");
+assert("\u1E9A".toUpperCase() == "A\u02BE");
+
+// Test full unicode range `0x0 - 0x10FFFF` and asserting with v8 engine.
+for (let i = 0; i <= 0x10FFFF; i++) {
+  let source = String.fromCodePoint(i);
+  let origLower = source.toLowerCase();
+  let origUpper = source.toUpperCase();
+  let code1: u64, code2: u64;
+
+  // collect all code points for lower case on AssemblyScript side
+  let origLowerCode = <u64>origLower.codePointAt(0);
+  if ((code1 = origLower.codePointAt(1)) >= 0) origLowerCode += <u64>code1 << 16;
+  if ((code2 = origLower.codePointAt(2)) >= 0) origLowerCode += <u64>code2 << 32;
+
+  // collect all code points for upper case on AssemblyScript side
+  let origUpperCode = <u64>origUpper.codePointAt(0);
+  if ((code1 = origUpper.codePointAt(1)) >= 0) origUpperCode += <u64>code1 << 16;
+  if ((code2 = origUpper.codePointAt(2)) >= 0) origUpperCode += <u64>code2 << 32;
+
+  // collect all code points for lower case on JavaScript side
+  let expectLowerCode = <u64>toLowerCaseFromIndex(i, 0);
+  if ((code1 = <i64>toLowerCaseFromIndex(i, 1)) >= 0) expectLowerCode += <u64>code1 << 16;
+  if ((code2 = <i64>toLowerCaseFromIndex(i, 2)) >= 0) expectLowerCode += <u64>code2 << 32;
+
+  // collect all code points for upper case on JavaScript side
+  let expectUpperCode = <u64>toUpperCaseFromIndex(i, 0);
+  if ((code1 = <i64>toUpperCaseFromIndex(i, 1)) >= 0) expectUpperCode += <u64>code1 << 16;
+  if ((code2 = <i64>toUpperCaseFromIndex(i, 2)) >= 0) expectUpperCode += <u64>code2 << 32;
+
+  if (origLowerCode != expectLowerCode) {
+    trace("origLowerCode != expectLowerCode", 3, i, <f64>origLowerCode, <f64>expectLowerCode);
+  }
+
+  if (origUpperCode != expectUpperCode) {
+    trace("origUpperCode != expectUpperCode", 3, i, <f64>origUpperCode, <f64>expectUpperCode);
+ }
+
+  assert(origLowerCode == expectLowerCode);
+  assert(origUpperCode == expectUpperCode);
+}
+
+
+
```

Comment:
All this block of tests transferred to `tests/compiler/std/string-casemapping.ts`. I guess it just appear after conflict merging

Reply:
I will check now, thanks @MaxGraey